### PR TITLE
homework8: done

### PR DESCRIPTION
**Pull request by Suren76**

### CODE REVIEW

File: src/test/java/homework8/demoqa/TestDroppable.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.testng.annotations.Test;
+
+import java.time.Duration;
+
+public class TestDroppable extends BaseTestDemoqa {
+    @Test
+    void testDragAndDrop() throws InterruptedException {
+        driver.get(BASEURL + "/droppable");
+
+        WebElement draggable = driver.findElement(By.id("draggable"));
```
Comment (David91) [inline src/test/java/homework8/demoqa/TestDroppable.java:18]: Are you sure `driver.findElement` will find and return the element?

File: src/test/java/homework8/expandtesting/BaseTestExpandtesting.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                return 31;
+            case 4:
+            case 6:
+            case 9:
+            case 11:
+                return 30;
+            case 2:
+                return 28;
+        }
+        return 50000;
```
Comment (David91) [inline src/test/java/homework8/expandtesting/BaseTestExpandtesting.java]: Why?

File: src/test/java/homework8/expandtesting/TestExpandtesting.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+import java.time.Duration;
+
+public class TestExpandtesting extends BaseTestExpandtesting {
+
+    @Test
+    void testSelectedData() {
+        driver.get(BASEURL + "/webpark");
+
+        Select dropDownMenu = new Select(driver.findElement(By.id("parkingLot")));
```
Comment (David91) [inline src/test/java/homework8/expandtesting/TestExpandtesting.java:19]: Are you sure your code will always work correctly without exceptions?

File: src/test/java/homework8/expandtesting/BaseTestExpandtesting.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        DateSplit s1 = new DateSplit(date1, "yyyy:MM:dd");
+        DateSplit s2 = new DateSplit(date2, "yyyy:MM:dd");
+        return getDateDelta(s1, s2);
+    }
+
+    protected double getTimeDelta(String time1, String time2) {
+        return Integer.parseInt(time2.split(":")[0]) - Integer.parseInt(time1.split(":")[0]) + (Integer.parseInt(time2.split(":")[1]) - Integer.parseInt(time1.split(":")[1]) >= 30 ? 0.5 : 0);
+    }
+
+    private int getMonthDaysCount(int monthNumber) {
```
Comment (aHarutPersonal) [inline src/test/java/homework8/expandtesting/BaseTestExpandtesting.java]: Why not use the Java Date class for this purpose?

File: src/test/java/homework8/expandtesting/BaseTestExpandtesting.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
+package homework8.expandtesting;
+
+import BaseTest.BaseTest;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BaseTestExpandtesting extends BaseTest {
+    final static String BASEURL = "https://practice.expandtesting.com";
```
Comment (aHarutPersonal) [inline src/test/java/homework8/expandtesting/BaseTestExpandtesting.java]: Please follow the naming conventions for constants
https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html
image

**Pull request merged**